### PR TITLE
sci-misc/boinc: Major updates and version bumps closing 4 open bugs

### PR DESCRIPTION
**Pull request by Yamakuzure**
This is an overhaul to make up some lost time due to far less spare time I had in the last couple of months than I've anticipated.

The updates are:
- New ebuild for boinc-7.4.52
  The other two series had no new release, yet.
- Gentoo-Bug: 587462
  Fix linker errors with wxGTK for boinc-7.6.33
- Gentoo-Bug: 493476
  Enable users to suspend/resume all projects without having to start
  and use the manager GUI.
- Gentoo-Bug: 584386 (partly)
  Fix "command substitution: ignored null byte in input" from bash-4.4+
- Gentoo-Bug: 593362
  Added a new shiny set of icons.

Further some cleanups, obsolete ebuilds and files have been removed.

Package-Manager: portage-2.3.2

@SoapGentoo @jlec : I do hope this is in order for you that I did not put all ebuild updates into separate PRs and/or commits, as they were basically the same. Apart from the new ebuild for boinc-7.4.52 of course, which has its own commit. I am very behind of my own schedule due to a very bad time management on my side, so I hope I can make up at least some of it.

### CODE REVIEW

**Comment by Yamakuzure** [comment]
Before someone else points this out:

I haven't checked wxwidgets.eclass thoroughly enough, and did not notice that is now supports EAPI 6.

If this is gonna be a problem, please let me know and I'll update the ebuilds to EAPI 6 and modify this PR.

**Comment by SoapGentoo** [comment]
@Yamakuzure could you still try to port it to `EAPI=6`, as wxwidgets is `EAPI=6` clean now?

**Comment by Yamakuzure** [comment]
Am Montag, 7. November 2016, 13:40:04 CET schrieb David Seifert:
> @Yamakuzure could you still try to port it to `EAPI=6`, as wxwidgets is
> `EAPI=6` clean now?
@SoapGentoo Almost done. However, repoman is not happy about the size of the 
new icon files.
I am a bit puzzled about what to do. The best solution for me would to remove 
the commit introducing the new icons, so they get removed.
Then I'll add a USE flag ("newicons" maybe?) that adds ImageMagick to the 
dependencies (CDEPEND?) and has src_install() to convert the source image I 
used to the target icons.
...or can they be mirrored somewhere?

**Comment by SoapGentoo** [comment]
@Yamakuzure don't worry about the images leave them as they are, rather focus on having the rest work. I'll take care of the images

**Comment by Yamakuzure** [comment]
@SoapGentoo @jlec : The update to EAPI 6 is done, and instead of pumping png icons into git, the -r1 ebuilds create all supported sizes using convert.

File: sci-misc/boinc/boinc-7.2.44-r1.ebuild [hunk 1]
```diff
@@ -99,7 +109,12 @@ src_install() {
 	keepdir /var/lib/${PN}
 
 	if use X; then
-		newicon "${S}"/packages/generic/sea/${PN}mgr.48x48.png ${PN}.png
+		# Create new icons. bug 593362
+		local SIZES="256 192 128 96 72 64 48 36 32 24 22 16"
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.2.44-r1.ebuild]: use bash arrays`=()` instead

File: sci-misc/boinc/boinc-7.2.44-r1.ebuild [hunk 2]
```diff
@@ -99,7 +109,12 @@ src_install() {
...
+		for s in $SIZES ; do
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.2.44-r1.ebuild]: make `s` local before

File: sci-misc/boinc/boinc-7.4.52-r1.ebuild [hunk 3]
```diff
@@ -76,9 +84,14 @@ src_prepare() {
 	# prevent bad changes in compile flags, bug 286701
 	sed -i -e "s:BOINC_SET_COMPILE_FLAGS::" configure.ac || die "sed failed"
 
+	# >=x11-libs/wxGTK-3.0.2.0-r3 has webview removed, bug 587462
+	epatch "${FILESDIR}"/fix_webview.patch
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.4.52-r1.ebuild]: use `PATCHES` instead and call `default` right at the beginning of src_prepare

File: sci-misc/boinc/boinc-7.2.44-r1.ebuild [hunk 4]
```diff
@@ -99,7 +109,12 @@ src_install() {
...
+			convert "${DISTDIR}"/${PN}.tif -resize ${s}x${s} "${WORKDIR}"/boinc_${s}.png || die
+			newicon -s $s "${WORKDIR}"/boinc_${s}.png boinc.png
+		done
 		make_desktop_entry boincmgr "${PN}" "${PN}" "Math;Science" "Path=/var/lib/${PN}"
 	fi
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.2.44-r1.ebuild]: check the `pkg_preinst`: you're assigning to group and group**s**, is this right?

Comment (Yamakuzure) [inline sci-misc/boinc/boinc-7.2.44-r1.ebuild]: Interesting... nobody ever fell over that one, it has been in there for ages...

**Comment by Yamakuzure** [comment]
@SoapGentoo Everything done as you suggested. Thank you very much (again!) for your invaluable help!

File: sci-misc/boinc/boinc-7.6.33-r1.ebuild [hunk 5]
```diff
@@ -117,7 +138,7 @@ pkg_preinst() {
 	# elog user about the need of being in video group
 	local groups="${PN}"
 	if use cuda; then
-		group+=",video"
+		groups+=",video"
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.33-r1.ebuild]: of course you need to fix this in all ebuilds 😄

File: sci-misc/boinc/boinc-7.6.33.ebuild [hunk 6]
```diff
@@ -2,11 +2,11 @@
 # Distributed under the terms of the GNU General Public License v2
 # $Id$
 
-EAPI=5
+EAPI=6
 
 WX_GTK_VER=3.0
 
-inherit autotools flag-o-matic linux-info systemd user versionator wxwidgets
+inherit autotools linux-info systemd user versionator wxwidgets
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.33.ebuild]: you're using `make_desktop_entry` without inheriting `eutils`

File: sci-misc/boinc/boinc-7.6.33-r1.ebuild [hunk 7]
```diff
@@ -99,7 +114,13 @@ src_install() {
 	keepdir /var/lib/${PN}
 
 	if use X; then
-		newicon "${S}"/packages/generic/sea/${PN}mgr.48x48.png ${PN}.png
+		# Create new icons. bug 593362
+		local s
+		local SIZES=(256 192 128 96 72 64 48 36 32 24 22 16)
+		for s in ${SIZES[@]} ; do
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.33-r1.ebuild]: this needs to be
`for s in "${SIZES[@]}" ; do`

File: sci-misc/boinc/boinc-7.6.33-r1.ebuild [hunk 8]
```diff
@@ -99,7 +114,13 @@ src_install() {
 	keepdir /var/lib/${PN}
 
 	if use X; then
-		newicon "${S}"/packages/generic/sea/${PN}mgr.48x48.png ${PN}.png
+		# Create new icons. bug 593362
+		local s
+		local SIZES=(256 192 128 96 72 64 48 36 32 24 22 16)
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.33-r1.ebuild]: compress this into one line:
`local s SIZES=( 16 22 24 32 36 48 64 72 96 128 192 256 )`
(notice that I sorted it by size)

File: sci-misc/boinc/boinc-7.6.33.ebuild (hunk 6 above)
Comment (Yamakuzure) [inline sci-misc/boinc/boinc-7.6.33.ebuild]: hmmm... right... versionator inherits it. I guess that's why there never was an error.

Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.33.ebuild]: exactly, relying on `versionator`'s implicit behaviour is not good

Comment (Yamakuzure) [inline sci-misc/boinc/boinc-7.6.33.ebuild]: done. eutils added.

File: sci-misc/boinc/boinc-7.6.33-r1.ebuild (hunk 8 above)
Comment (Yamakuzure) [inline sci-misc/boinc/boinc-7.6.33-r1.ebuild]: done.

File: sci-misc/boinc/boinc-7.6.33-r1.ebuild (hunk 7 above)
Comment (Yamakuzure) [inline sci-misc/boinc/boinc-7.6.33-r1.ebuild]: done.

(I knew there was something odd, but couldn't tell what... :) )

**Comment by Yamakuzure** [comment]
@SoapGentoo Hopefully everything is up to scratch now. Sorry for the delay. I wanted to finish yesterday, but ran out of time. Had to catch my train.

**Comment by SoapGentoo** [comment]
@Yamakuzure I did some minor touchups in the ebuilds. Thanks for the work.

**Comment by Yamakuzure** [comment]
@SoapGentoo *yay* ! Thank you very much! I have reviewed your "touchups", and I think we are close to having eradicated all traces of the ancient ebuilds. :-) And I am very happy about being happy to close those bugs, yes, finally! (\me is happy!)

**Pull request closed without merging**